src/components/presentationals/ProfileSettings/ProfileSettings.jsx

```diff
@@ -0,0 +1,162 @@
+
+    const reader = new FileReader();
+    reader.addEventListener('load', () => {
+      previewer.src = reader.result;
+    });
+    reader.readAsDataURL(image);
+    toast.success('Photo Ready to be saved to profile', 'DONE');
+    this.setState({ previewImage: image });
+  };
+
+  render() {
+    return (
+      <div className="settings">
+        <div className="edit-profile">
+          {this.state.isLoading && (
+            <Loader text="Updating Profile" size="large teal" />
+          )}
+          <Form onSubmit={this.handleSubmit}>
+            <Form.Group widths="equal">
+              <Form.Input
+                fluid
+                label="First name"
+                placeholder="Jane"
+                defaultValue={this.props.firstName}
+                name="firstName"
+              />
+              <Form.Input
+                fluid
+                label="Last name"
+                placeholder="Doe"
+                defaultValue={this.props.lastName}
+                name="lastName"
+              />
+            </Form.Group>
+            <Form.Group widths="equal">
+              <Form.Input
+                fluid
+                label="Email Address"
+                placeholder="example@mail.com"
+                defaultValue={this.props.email}
+                type="email"
+                name="email"
+              />
+              <Form.Input
+                fluid
+                label="Username"
+                placeholder="janedoe1"
+                defaultValue={this.props.username}
+                name="username"
+              />
+            </Form.Group>
+            <Form.TextArea
+              label="Bio"
+              name="bio"
+              placeholder="Tell us more about you...(450 characters max)"
+              maxLength="450"
+              defaultValue={this.props.bio}
+              rows={5}
+            />
+            <div
+              className="field"
+              style={{
+                display: 'flex',
+                justifyContent: 'flex-end',
+              }}
+            >
+              <Button type="submit" value="SAVE CHANGES" className="btn-dark" />
+              <Form.Input
+                fluid
+                type="file"
+                accept="image/*"
+                name="upload_profile_picture"
+                placeholder="Select Profile Pix"
+                onChange={this.previewImage}
+              />
+            </div>
+          </Form>
+        </div>
+      </div>
+    );
+  }
+}
+
+ProfileSettings.propTypes = {
+  currentView: PropTypes.func,
+  uploadImageToServer: PropTypes.func.isRequired,
+  updateUserProfileRequest: PropTypes.func.isRequired,
+  firstName: PropTypes.string,
+  lastName: PropTypes.string,
+  username: PropTypes.string,
+  email: PropTypes.string,
+  updatedAt: PropTypes.string,
+  bio: PropTypes.string,
+};
+
+const mapStateToProps = state => {
+  const { data } = state.profile.profile;
+  return data;
```

Comment:
I think you have to return the actual data you want to map your state to from the parent provided prop for the prop validation to work well

```javascript
mapStateToProps = state => ({
profile: {
  currentView: state.profile.profile.currentView,
  uploadImageToServer: state.profile.profile.,
  updateUserProfileRequest: state.profile.profile.updateUserProfileRequest,
  firstName: state.profile.profile.firstName,
  lastName: state.profile.profile.lastName,
  username: state.profile.profile.username,
  email: state.profile.profile.email,
  updatedAt: state.profile.profile.updatedAt,
  bio: state.profile.profile.bio,
}
});
```

Reply:
ok, thanks